source/generators/SingleParticleGenerator.cc

```diff
@@ -62,11 +61,11 @@ costheta_max_(1.), phi_min_(0.), phi_max_(2.*pi)
   msg_->DeclareProperty("region", region_,
     "Set the region of the geometry where the vertex will be generated.");
 
-  msg_->DeclareProperty("momentum_X", momentum_X_,
+  msg_->DeclareProperty("momentum_X", momentum_[0],
```

Comment:
We could simplify this using the function `DeclarePropertyWithUnit`, which accepts 3-vectors (see https://geant4.kek.jp/lxr/source/intercoms/src/G4GenericMessenger.cc#L106).